Remove requirement for separate Qt installation

jonathanperret opened the pull request:
The only reason to require a standalone Qt installation was the `lrelease` tool used to convert the translation files.

It turns out PySide6, which we install through `pip`, already includes this tool under the name `pyside6-lrelease`.

This PR changes the call from `lrelease` to `pyside6-lrelease`, and removes the installation of Qt from the GitHub Actions workflow and from the development setup instructions.

Except on Linux, where the Qt libraries which are bundled in PySide6 have themselves dependencies on system libraries such as `libgl1` and a dozen others. In that case it seemed simpler to keep installing a system Qt 6 package to pull these system dependencies in, even though none of the Qt components installed system-wide will be used.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Upgraded Python environment setup to the latest version, enhancing compatibility and performance.
  - Streamlined installation process for Qt6 system dependencies for better efficiency.

- **Documentation**
  - Improved clarity and conciseness of the installation instructions in the README, focusing on essential steps for users.

- **Chores**
  - Updated command for generating binary files to ensure compatibility with newer versions of the PySide6 framework.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jonathanperret commented:
Since the Build workflow doesn’t trigger for PRs here’s a run on my fork on the same commit, to demonstrate everything works: https://github.com/jonathanperret/ayab-desktop/actions/runs/9908222560

X-sam reviewed (changes requested):
Can you break the changes out into multiple PRs? I'm ok removing QT, but I am less prepared for testing the updates to actions at the moment, and I added the type generation for a reason so I would rather we don't remove it without further discussions and testing.

jonathanperret commented:
> Can you break the changes out into multiple PRs? I'm ok removing QT, but I am less prepared for testing the updates to actions at the moment, and I added the type generation for a reason so I would rather we don't remove it without further discussions and testing.

You're right, I got a bit carried away there. I've pared down the PR to just the removal of Qt. As I added in the PR description, on Linux we need to somehow get Qt's system dependencies installed, so I left `apt install -y qt6-base-dev qt6-tools-dev-tools` in to get these.

dl1com commented:
@X-sam as @jonathanperret did some changes to the PR, do you think we can merge it now?
